icu4c/source/i18n/calendar.cpp

```diff
@@ -1927,6 +1927,10 @@ void Calendar::roll(UCalendarDateFields field, int32_t amount, UErrorCode& statu
 
             // Now roll between start and (limit - 1).
             int32_t gap = limit - start;
+            if (gap == 0) {
+                status =  U_INTERNAL_PROGRAM_ERROR;
+                return;
+            }
```

Comment:
Well, this'll keep it from crashing, at least...

But no idea how we're actually getting into this state?  Is it worth trying to figure that out, or do you think it's rare enough not to worry that much about?

Reply:
The algorithm is very complex here and the case cause this issue that I can find to reproduce is in many many years ago ( about ~1120000 years ago )